src/status_im2/contexts/communities/menus/community_options/view.cljs

```diff
@@ -82,19 +82,19 @@
   {:icon                :i/log-out
    :label               (i18n/label :t/leave-community)
    :accessibility-label :leave-community
-
    :danger?             true
    :on-press            #(rf/dispatch [:bottom-sheet/show-sheet
                                        {:content        (constantly [leave-menu/leave-sheet id])
                                         :content-height 400}])})
 
 (defn cancel-request-to-join
   [id]
-  {:icon                :i/block
-   :label               (i18n/label :t/cancel-request-to-join)
-   :accessibility-label :cancel-request-to-join
-   :danger?             true
-   :on-press            #(js/alert (str "implement action" id))})
+  (let [request-id (rf/sub [:communities/my-pending-request-to-join id])]
```

Comment:
should this `sub` be higher up the chain?

Reply:
I think the answer is yes, the subscription should be higher up, in an actual view component that needs re-rendering capabilities. Even though the code above works, I believe subscriptions should be used for building up hiccup structures/components, not normal Clojure data structures.